xds/src/main/java/io/grpc/xds/ClusterImplLoadBalancer.java

```diff
@@ -274,6 +283,28 @@ private List<EquivalentAddressGroup> withAdditionalAttributes(
       return newAddresses;
     }
 
+    private ClusterLocality createClusterLocalityFromAttributes(Attributes addressAttributes) {
+      Locality locality = addressAttributes.get(InternalXdsAttributes.ATTR_LOCALITY);
+      String localityName = addressAttributes.get(InternalXdsAttributes.ATTR_LOCALITY_NAME);
+
+      // Endpoint addresses resolved by ClusterResolverLoadBalancer should always contain
+      // attributes with its locality, including endpoints in LOGICAL_DNS clusters.
+      // In case of not (which really shouldn't), loads are aggregated under an empty
+      // locality.
+      if (locality == null) {
```

Comment:
It shouldn't be, but is possible that locality is set but locality name is null.  You should probably have an else clause that does a null check on localityName.

Reply:
This is existing code, and it seems in general code assumes the values to be non-null. I'd much rather we handle that centrally in `io.grpc.xds.client.Locality.create()` than lots of random not-possible-to-trigger checks. The data is primarily coming from a proto, so it will be `""` when unset.

Reply:
Are you suggesting to move the locality name null check to `io.grpc.xds.client.Locality.create()`?

Reply:
Oh, I see now Larry's comment wasn't about the Locality struct, but instead the attribute. That had been discussed when the code was originally introduced:
https://github.com/grpc/grpc-java/pull/11133#discussion_r1586584776

This PR is likely to be reverted (in some way) later, once the old PF policy goes away. So changes to the existing code are likely to be lost.